### DISCUSSION TITLE
Proper way to user custom fonts (Calibri, Courier New..) in mpdf 8?

### DISCUSSION THREAD

**Discussion by Laci357** [mpdf/mpdf#1559] [Q&A] [answered]
Hi!
Is there any way in mpdf 8 to add custom fonts (not included in the mpdf library by default) in a similar way as it was in mdpf7? 
I've tried to incorporate the following example in my code:
https://mpdf.github.io/fonts-languages/fonts-in-mpdf-7-x.html

However, FontVariables() does not seem to exist in mpdf8. 
A snippet of my code is:

```
$mpdf = new \Mpdf\Mpdf([
	'mode' => 'utf-8',
	'format' => 'A4',
	'fontDir' => [
	\_\_DIR\_\_.'/CustomFonts/',
	\_\_DIR\_\_.'/../vendor/mpdf/mpdf/ttfonts/'
	],
	'onlyCoreFonts' => true,
	'default_font_size' => '13',
	'tempDir' => \_\_DIR\_\_ . '/temp'
]);

$mpdf->AddFont('Calibri','R');
$mpdf->default_font('Calibri');
```

The response I recieve:

> Fatal error: Uncaught Mpdf\MpdfException: Font "calibry - R" is not supported in /var/www/tools/vendor/mpdf/mpdf/src/Mpdf.php:3848 Stack trace: #0 /var/www/tools/pdfgenerator/pdf_generator.php(477): Mpdf\Mpdf->AddFont() #1 /var/www/tools/pdfgenerator/pdf_generator.php(990): printMpdf() #2 {main} thrown in /var/www/tools/vendor/mpdf/mpdf/src/Mpdf.php on line 3848

Any help is much appreciated.

**Comment by finwe** [comment, marked as answer]
Using custom fonts in 8.x is literally the same as in 7.x. Just use ALL the example code, especially `fontdata` portion of mPDF config. Also note that the font name has to be defined as lowercase.

**Reply by finwe** [reply]
And, of course, FontVariables class still exists. https://github.com/mpdf/mpdf/blob/development/src/Config/FontVariables.php
